app/controllers/search_controller.rb

```diff
@@ -8,10 +8,10 @@ def create
     coordinates = get_lat_lon(cookies["coordinates"] || params["search_location"])
     cookies.delete "coordinates";
     return handle_invalid_location(coordinates) unless coordinates
-
+    activities = params["activities"].map { |p| eval(p).values}.flatten
     session[:search] = {
```

Comment:
`eval(p)` - I'll look up what this does but I'm interested the reasoning for using it